src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -650,16 +654,24 @@ fn expand_files_helper(
       Some((dir, _metadata)) => {
         {
           let mut files_unlocked = files.lock();
-          for file in dir.get_files() {
-            let file_digest: Result<Digest, String> = file.get_digest().try_into();
+          for file in &dir.files {
+            let file_digest: Result<Digest, String> = file
+              .digest
+              .as_ref()
+              .map(|d| d.try_into())
+              .unwrap_or(Ok(EMPTY_DIGEST));
```

Comment:
I did try `impl TryFrom<Option<bazel_protos::Digest>> for hashing::Digest` but it fails due to limitations in `std` and the current Rust stable versions:

```
error[E0119]: conflicting implementations of trait `std::convert::TryFrom<std::option::Option<gen::build::bazel::remote::execution::v2::Digest>>` for type `hashing::Digest`:
  --> process_execution/bazel_protos/src/conversions.rs:46:1
   |
46 | impl TryFrom<Option<crate::gen::build::bazel::remote::execution::v2::Digest>> for hashing::Digest {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T, U> TryFrom<U> for T
             where U: Into<T>;
   = note: upstream crates may add a new impl of trait `std::convert::From<std::option::Option<gen::build::bazel::remote::execution::v2::Digest>>` for type `hashing::Digest` in future versions

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> process_execution/bazel_protos/src/conversions.rs:46:1
   |
46 | impl TryFrom<Option<crate::gen::build::bazel::remote::execution::v2::Digest>> for hashing::Digest {
   | ^^^^^------------------------------------------------------------------------^^^^^---------------
   | |    |                                                                            |
   | |    |                                                                            `hashing::Digest` is not defined in the current crate
   | |    `std::option::Option` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead
```

I could move all of the conversions to `hashing` to try and avoid the cycle between the crates. But there is a generic implementation for `TryFrom` that conflicts with any other implementations, and we would apparently need specialization in stable and not nightly to be able to do the Option<Digest> bit to avoid the catch-all generic impl.  https://github.com/rust-lang/rust/issues/50133

Reply:
Oh well, a method is just as usable :) Thanks for trying!

Reply:
Will go with a method, I actually had already introduced one called `to_pants_digest_opt` as part of these "half one way, half the other way" solutions.